### DISCUSSION TITLE
Computing energy

### DISCUSSION THREAD

**Discussion by lukanov-9b** [brucefan1983/GPUMD#674] [Q&A]
Hello, dear Zheyong Fan!
 
Could you tell me please, how can I compute energies of structures separately? I created a model.xyz-file with structure configurations (each structure about 64 atoms). After that I used nep-potential (nep.txt) file for computing, indicating prediction mode as 1. But I have some results:
(First column - prediction, second column - real values)
-12.2512 -21.6235
-14.1193 -21.2266
-14.2669 -20.5895
-14.0811 -20.6964
-12.0444 -20.6308
-15.2373 -19.5306
-13.6627 -19.9599
-11.9918 -20.9433
-13.6622 -19.9643
-13.6825 -19.9623

Structures were taken from test.xyz file used for training process. I can't understand, why errors in this case, when I predict energies separately from training, are significant. Thanks!

**Comment by brucefan1983** [comment]
It's not very clear from your description what you did exactly. I guess you mean you used the `prediction 1` keyword in the `nep.in` file to predict results for a `train.xyz` file, where the structures in `train.xyz` are some new structures you want to predict?

**Comment by brucefan1983** [comment]
If so, you can post figures showing the training and testing results. Also the `nep.in` file. The more information the better.

**Comment by lukanov-9b** [comment]
So, I have some structures and I need to know energies of them. I read that I can use nep-code for this purpose if I have nep.txt file in the directory and prediction 1 key-word. I wanted to check this on structures that were used ifor test.xyz-file, when I trained nep-potential. And I have different errors...

**Comment by lukanov-9b** [comment]
When I trained model 
![test](https://github.com/brucefan1983/GPUMD/assets/124994372/ea95ef10-43e1-4859-8ea9-3d457c9f77fc)
![train](https://github.com/brucefan1983/GPUMD/assets/124994372/ea5c4844-44fd-4bcf-b84d-dd1156171aee)

If I do this from prediction 1, aI have
![image](https://github.com/brucefan1983/GPUMD/assets/124994372/7da879db-b68b-402b-84be-7cd7fcf27f63)

**Comment by brucefan1983** [comment]
Could you please post the two `nep.in` files, one for training, one for testing (with `prediciton 1`).

Also, could you tell me the versions of GPUMD if you did them using different versions.

**Comment by lukanov-9b** [comment]
The first nep.in file:
type    5 Hf Nb Ta Ti Zr


The second nep.in file:
type    5 Hf Nb Ta Ti Zr
prediction 1

**Comment by lukanov-9b** [comment]
I have Version 3.8

**Comment by brucefan1983** [comment]
Then you can send me the inputs to debug (via email).

**Comment by brucefan1983** [comment]
You must have used different versions of GPUMD for training and prediction and the two versions happen to  have different default values for `basis_size`. 

If you add `basis_size 12 12` to your `nep.in` and do the predition again, you will get the correct results.

**Comment by brucefan1983** [comment]
I got `basis_size 12 12` from your `nep.txt`. The point is that `nep.in` (with many default parameters that might be changed from version to version) should be consistent with `nep.txt` to give correct predictions.

**Comment by lukanov-9b** [comment]
Hello, dear Zheyong Fan! It has worked, thank you so much!!!
 
I wanted ask you about one moment of MC. Could you tell me, please, what difference are between 
<run ...> and <mc_trials>
 
For example, I use this configuration:
mc canonical 1 1000 500 500
time_step 0
dump_exyz 1000
run 1000000
 
GPUMD does 1000 steps of MC in each step (from 1000000 in run-option)? When I used mc canonical 1 1 500 500 I got 1 or 0 in mcmd.out. But when I use more then 1 mc trials, I have float values in mcmd.out.

**Reply by brucefan1983** [reply]
Hello, I think you can understand everything by reading the following two pages:

* https://gpumd.org/gpumd/input_parameters/mc.html

* https://gpumd.org/gpumd/output_files/mcmd_out.html#mcmd-out
